commit 00c045e95ec0c861c55092b60cb484844dba7521 (refs/changes/33/3533/2)
Author: Jason King <jason.king@joyent.com>
Date:   2018-03-03T00:07:58-06:00 (1 year, 7 months ago)
    
    OS-6733 SADB_EXPIRE messages omit KMC extension
    Reviewed by: Dan McDonald <danmcd@joyent.com>

diff --git a/usr/src/uts/common/inet/ip/sadb.c b/usr/src/uts/common/inet/ip/sadb.c
index 40d5078526..44ebb21db3 100644
--- a/usr/src/uts/common/inet/ip/sadb.c
+++ b/usr/src/uts/common/inet/ip/sadb.c
@@ -3767,7 +3767,8 @@ sadb_expire_assoc(queue_t *pfkey_q, ipsa_t *assoc)
 	}
 
 	alloclen = sizeof (*samsg) + sizeof (*current) + sizeof (*expire) +
-	    2 * sizeof (sadb_address_t) + sizeof (*saext);
+	    2 * sizeof (sadb_address_t) + sizeof (*saext) +
+	    sizeof (sadb_x_kmc_t);
 
 	af = assoc->ipsa_addrfam;
 	switch (af) {
@@ -3896,6 +3897,10 @@ sadb_expire_assoc(queue_t *pfkey_q, ipsa_t *assoc)
 		ASSERT(mp->b_wptr != NULL);
 	}
 
+	mp->b_wptr = sadb_make_kmc_ext(mp->b_wptr, end, assoc->ipsa_kmp,
+	    assoc->ipsa_kmc);
+	ASSERT(mp->b_wptr != NULL);
+
 	/* Can just putnext, we're ready to go! */
 	putnext(pfkey_q, mp1);
 }
